### ISSUE TITLE
Testing wallet generation, restoration with BIP39 give me a Wallet sanity check failed error

### ISSUE THREAD

**Issue by ElucGeek** [ElementsProject/lightning#7491]
I have installed core lightning, let the hsm_secret automatically generated, then encrypted it. I can run the node, it's all fine.

Now I want to dismiss this wallet and create a new one from a BIP39 mnemonic phrase.

I stop lightning, I rename hsm_secret (mv hsm_secret_backup).
Then I do:
hsmtool generatehsm hsm_secret
Enter 12 words generated from Sparrow wallet (for testing purpose).
hsmtool encrypt hsm_secret
give a password

Then I run lightingd again I get this error:
lightningd: Wallet sanity check failed.

If I delete the new hsm_secret and mv back from the backup, it's running fine again.

What did I do wrong? Is there more than I need to delete (move) to create this new wallet? There is no channel, no fund yet, I'm just testing wallet generation and restoration to be sure I end up with the same on-chain address in the end.

Once I'm confident with the process I will generate a proper BIP39 from an offline device, make by backup and start over, generate, delete, restore, then add funds, to be sure everything is fine before creating my first channels. If you have any other advise for this process, let me know.

**Comment by dennisreimann** [comment]
> however the hkdf function we use to go from root secret to node_id, and then to the on-chain addresses is not a derivation that sparrow, or any other wallet appear to support at the moment.

Does this also mean that the descriptors yielded by `lightning-hsmtool dumponchaindescriptors` won't work when imported into e.g. Sparrow? I'd like to import my Core Lightning onchain wallet as watch-only wallet into Sparrow and I can import the descriptors, but they don't yield any transaction though some should be present. /cc @craigraw

**Comment by Lagrang3** [comment]
> > however the hkdf function we use to go from root secret to node_id, and then to the on-chain addresses is not a derivation that sparrow, or any other wallet appear to support at the moment.
> 
> Does this also mean that the descriptors yielded by `lightning-hsmtool dumponchaindescriptors` won't work when imported into e.g. Sparrow? I'd like to import my Core Lightning onchain wallet as watch-only wallet into Sparrow and I can import the descriptors, but they don't yield any transaction though some should be present. /cc [@craigraw](https://github.com/craigraw)

You can create a watch-only wallet on Sparrow with the descriptors dumped by `lightning-hsmtool`.
That's something I have used myself. There is a small caveat though:
Sparrow will not let you import the descriptor out of the box like Bitcoin's wallet would and you will have to manually take the extended key `xpub...` and derive from it this path `m/0` and then feed that into Sparrow and then Sparrow will correctly generate all keys from that. Inserting `xpub...` and setting a custom derivation path `m/0` won't work, Sparrow will try to populate your wallet with addresses `xpub.../0/*`  and `xpub.../1/*` (change) instead of `xpub.../0/0/*`.
Also it should be noted that, by looking at the descriptors, the change addresses `xpub.../0/1/*` are not used by
core-lightning.

**Comment by Lagrang3** [comment]
Did you scan all three descriptors?

**Comment by dennisreimann** [comment]
Yes, but the Nested Segwit one doesn't have any transactions, which is expected I think. I've also increased the gap limit, which didn't help.

The output of `lightning-cli bkpr-listaccountevents wallet` has some transactions which I cannot find in the Taproot and Native Segwit wallets, which I've imported with the derived xpub. It looks like these are from force-closed channels.

btw for anyone coming across this and trying to derive the xpub themselves: I've used the Python btclib like this:

```python
#!/usr/bin/env python3
from btclib.bip32.bip32 import derive
print(derive("xpub...", "m/0"))
```

**Comment by craigraw** [comment]
> It'd be nice if Sparrow would support the import directly

Is there some standard being used here? Sparrow follows BIP32, I'm unaware of a standard that uses `xpub.../0/0/*`.

**Comment by Lagrang3** [comment]
Agree, it is not standard. But wallet descriptors are standardized in BIP-380.
I love Sparrow's interface. I wish it had at least partial support for descriptors, so that instead
of importing a wallet from the xpub or xpriv, one might paste a descriptor and it would automagically know
what to do with it.

**Comment by craigraw** [comment]
I've now added support for deriving to maximum BIP32 account level where the child path in the output descriptor contains more than two elements.

>  I wish it had at least partial support for descriptors, so that instead
of importing a wallet from the xpub or xpriv, one might paste a descriptor and it would automagically know
what to do with it.

You can enter a wide variety of output descriptors when creating a wallet, including `xprv`s. File > New Wallet > Edit button on the right in the Settings tab. Sparrow will need to coerce it to a standard BIP32 wallet with receive and change chains, but otherwise it tries to be accommodative.
